functorch/__init__.py

```diff
@@ -91,7 +91,15 @@ def _functorch_str(tensor):
         return _old_str(tensor)
 
     value = _C.get_unwrapped(tensor)
+    # Disable temporarily kDynamicLayerFrontModeKey/kDynamicLayerBackModeKey as included dispatch keys
+    dl_enabled = _C.tls_set_is_included()
+    if (dl_enabled):
+        _C._set_dynamic_layer_keys_included(False)
     value_repr = repr(value)
+    # Reenable kDynamicLayerFrontModeKey/kDynamicLayerBackModeKey as included dispatch keys
+    if (dl_enabled):
+        _C._set_dynamic_layer_keys_included(True)
+
```

Comment:
@vfdev-5 I'm a bit confused about why our current tests pass ([1](https://github.com/pytorch/functorch/blob/b2d176daec0cfaee7d047a77d63cfc1ef020e057/test/test_vmap.py#L986-L1002), [2](https://github.com/pytorch/functorch/blob/main/test/test_eager_transforms.py#L723-L740)). Do you have some insight into that?

Reply:
I see, the problem is the printing path does different things on large tensor vs small tensors

Reply:
Yes, I'm also surprised that test_eager_transforms.py::test_tensor_print was passing before this PR. I'll check that.
As for vmap, IMO they should always pass as it does not modify dispatch keys (tls_set_dispatch_key_included).

Reply:
Seems like there is no printing issue for scalars but there is an issue with non-scalar tensors.